### PR TITLE
feat: add support for running application in subpath

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
https://github.com/user-attachments/assets/12794fb2-c887-465c-a71e-b079deca9263

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/package.json [hunk 1]
```diff
@@ -199,6 +199,7 @@
     "cypress": "12.17.4",
     "cypress-postgresql": "^1.0.8",
     "dotenv": "^16.0.0",
+    "dotenv-webpack": "^8.1.0",
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/package.json:202]: Do we need this package?

File: openmetadata-ui/src/main/resources/ui/src/utils/AuthProvider.util.ts [hunk 2]
```diff
@@ -42,17 +42,21 @@ export const EXPIRY_THRESHOLD_MILLES = 25 * 1000;
 export const EXPIRY_THRESHOLD_MILLES_PUBLIC = 2 * 60 * 1000;
 
 export const getRedirectUri = (callbackUrl: string) => {
+  const subPath = process.env.APP_SUB_PATH ?? '';
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/AuthProvider.util.ts:45]: We can declare it at file scope and use it across instead querying at each step

File: openmetadata-ui/src/main/resources/ui/src/utils/AuthProvider.util.ts [hunk 3]
```diff
@@ -42,17 +42,21 @@ export const EXPIRY_THRESHOLD_MILLES = 25 * 1000;
...
-    ? 'http://localhost:3000/callback'
+    ? `http://localhost:3000${subPath}/callback`
     : !isNil(callbackUrl)
     ? callbackUrl
-    : `${window.location.origin}/callback`;
+    : `${window.location.origin}${subPath}/callback`;
 };
 
 export const getSilentRedirectUri = () => {
+  const subPath = process.env.APP_SUB_PATH ?? '';
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/AuthProvider.util.ts:55]: same

File: openmetadata-ui/src/main/resources/ui/webpack.config.dev.js [hunk 4]
```diff
@@ -249,6 +252,11 @@ module.exports = {
     // Route all requests to index.html so that app gets to handle all copy pasted deep links
     historyApiFallback: {
       disableDotRule: true,
+      ...(process.env.APP_SUB_PATH
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/webpack.config.dev.js:255]: Let's just declare it once and re-use

**Review by chirag-madlani — COMMENTED** [review]
Added some minor nit

File: openmetadata-ui/src/main/resources/ui/package.json (hunk 1 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/package.json:202]: Yes, we need to load the env

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsight/EmptyGraphPlaceholder.tsx [hunk 5]
```diff
@@ -37,10 +36,11 @@ export const EmptyGraphPlaceholder = ({ icon }: { icon?: ReactElement }) => {
         <Transi18next
           i18nKey="message.refer-to-our-doc"
           renderElement={
-            <Link
+            <a
+              href={DATA_INSIGHT_DOCS}
               rel="noreferrer"
               target="_blank"
-              to={{ pathname: DATA_INSIGHT_DOCS }}
+              title="Data Insight Documentation"
```
[the remark this answers is not in the document]
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/DataInsight/EmptyGraphPlaceholder.tsx:43]: Thanks @chirag-madlani , I will take care of these in my next patch.

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**